python/ray/data/_internal/plan.py

```diff
@@ -499,7 +501,7 @@ def execute_to_iterator(
             block_iter = itertools.chain([next(gen)], gen)
         except StopIteration:
             pass
-        return block_iter, executor.get_stats()
+        return block_iter, executor.get_stats(), executor
```

Comment:
The second component is also from executor. It'd look better to return a ExecutionContext to wrap them and to accommodate the future needs.

Reply:
Ok this turned out to be a bit messy and IMO not worth it for a temporary state of things--- how about we refactor with removing the legacy plan class entirely?

Reply:
Yeah sounds good.
One more issue is this just returns an iterator of block refs, so if we want to eagerly release block memory after consumption, we'll need to add `owned` bit (RefBundle doesn't have this issue).